rust/connlib/snownet/src/node.rs

```diff
@@ -140,6 +140,12 @@ where
         (&self.private_key).into()
     }
 
+    pub fn is_connected_to(&self, key: PublicKey) -> bool {
+        self.connections
+            .iter_established()
+            .any(|(_, c)| c.remote_pub_key == key && c.tunnel.time_since_last_handshake().is_some())
```

Comment:
Does `time_since_last_handshake` return `None` if the handshake has timed out?

Reply:
This is O(n) with the number of established connections but I assume that's always low, and this isn't called much?

Reply:
> Does `time_since_last_handshake` return `None` if the handshake has timed out?

Yeah as far as I understand boringtun, yes. It is unfortunately the only API they expose to detect whether the Tunnel has handshaked.